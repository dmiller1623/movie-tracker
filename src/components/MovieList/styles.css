.movie-cards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center; }

.more-btn {
  width: 20rem;
  margin-bottom: 2rem;
  border-radius: 10px; }

@media only screen and (max-width: 430px) {
  .more-btn {
    width: 15rem; } }
