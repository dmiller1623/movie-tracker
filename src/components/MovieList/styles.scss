main {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.more-btn {
  width: 342px;
  height: 500px;
  text-align: center;
  color: white;
  border: 2px solid white;
  border-radius: 15px;
}

.more-text {
  margin: 170px 20px;
  border-radius: 80px;
  border: 2px solid white;
  padding: 50px;
  font-size: 2rem;
  cursor: pointer;
}