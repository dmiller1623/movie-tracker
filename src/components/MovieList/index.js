import React from 'react';
import MovieCard from '../../containers/MovieCard';
import './styles.css';

const MovieList = ({ movies, favorites = [] }) => {
  const displayedMovies = movies.map( (movie, index) => {
    const starColor = favorites.includes(movie.movie_id) ? '#ffd24d' : '#f2f2f2';
    return <MovieCard {...movie} key={index} starColor={starColor}/>
  });

  return (
    <main>
      {displayedMovies}
    </main>
  )
}

export default MovieList;
