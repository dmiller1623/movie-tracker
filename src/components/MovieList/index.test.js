import React from 'react';
import { shallow } from 'enzyme';
import MovieList from '../MovieList';

describe('MovieList', () => {

  it('should match the snapshot with no movies', () => {
    const movies = [];
    const wrapper = shallow(<MovieList movies={movies} />)

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should match the snapshot with one movie', () => {
    const movies = [{
      movie_id: 0,
      title: 'test',
      poster_path: 'test',
      release_date: 2018-9-1,
      vote_average: 1,
      overview: 'test'
    }];
    const wrapper = shallow(<MovieList movies={movies} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with multiple movies', () => {
    const movies = [{
      movie_id: 0,
      title: 'test1',
      poster_path: 'test1',
      release_date: 2018-9-1,
      vote_average: 1,
      overview: 'test1'
    }, {
      movie_id: 1,
      title: 'test2',
      poster_path: 'test2',
      release_date: 2018-9-1,
      vote_average: 1,
      overview: 'test2'
    }];
    const wrapper = shallow(<MovieList movies={movies} />)

    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should change starColor based on if the movie is included in favorites', () => {
    const movies = [{
      movie_id: 0,
      title: 'test',
      poster_path: 'test',
      release_date: 2018-9-1,
      vote_average: 1,
      overview: 'test'
    }];
    let favorites = [];
    let wrapper = shallow(<MovieList movies={movies} favorites={favorites}/>)
    
    expect(wrapper.find('path').prop('fill')).toEqual('#f2f2f2');

    favorites = [{
      movie_id: 0,
      title: 'test',
      poster_path: 'test',
      release_date: 2018-9-1,
      vote_average: 1,
      overview: 'test'
    }];
    wrapper = shallow(<MovieList movies={movies} favorites={favorites}/>)

    expect(wrapper.find('path').prop('fill')).toEqual('#ffd24d');
  });

});