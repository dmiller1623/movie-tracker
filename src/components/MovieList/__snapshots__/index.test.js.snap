// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList should match the snapshot with multiple movies 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieList
    favorites={
        Array [
            Object {
              "movie_id": 0,
              "overview": "test1",
              "poster_path": "test1",
              "release_date": 2008,
              "title": "test1",
              "vote_average": 1,
            },
            Object {
              "movie_id": 1,
              "overview": "test2",
              "poster_path": "test2",
              "release_date": 2008,
              "title": "test2",
              "vote_average": 1,
            },
          ]
    }
    movies={
        Array [
            Object {
              "movie_id": 0,
              "overview": "test1",
              "poster_path": "test1",
              "release_date": 2008,
              "title": "test1",
              "vote_average": 1,
            },
            Object {
              "movie_id": 1,
              "overview": "test2",
              "poster_path": "test2",
              "release_date": 2008,
              "title": "test2",
              "vote_average": 1,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-cards"
>
          <Connect(MovieCard)
                    movie_id={0}
                    overview="test1"
                    poster_path="test1"
                    release_date={2008}
                    starColor="#f2f2f2"
                    title="test1"
                    vote_average={1}
          />
          <Connect(MovieCard)
                    movie_id={1}
                    overview="test2"
                    poster_path="test2"
                    release_date={2008}
                    starColor="#f2f2f2"
                    title="test2"
                    vote_average={1}
          />
</div>,
        <button
          className="more-btn"
          onClick={[Function]}
>
          View More
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(MovieCard)
              movie_id={0}
              overview="test1"
              poster_path="test1"
              release_date={2008}
              starColor="#f2f2f2"
              title="test1"
              vote_average={1}
/>,
            <Connect(MovieCard)
              movie_id={1}
              overview="test2"
              poster_path="test2"
              release_date={2008}
              starColor="#f2f2f2"
              title="test2"
              vote_average={1}
/>,
          ],
          "className": "movie-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "movie_id": 0,
              "overview": "test1",
              "poster_path": "test1",
              "release_date": 2008,
              "starColor": "#f2f2f2",
              "title": "test1",
              "vote_average": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "movie_id": 1,
              "overview": "test2",
              "poster_path": "test2",
              "release_date": 2008,
              "starColor": "#f2f2f2",
              "title": "test2",
              "vote_average": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View More",
          "className": "more-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View More",
        "type": "button",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-cards"
>
            <Connect(MovieCard)
                        movie_id={0}
                        overview="test1"
                        poster_path="test1"
                        release_date={2008}
                        starColor="#f2f2f2"
                        title="test1"
                        vote_average={1}
            />
            <Connect(MovieCard)
                        movie_id={1}
                        overview="test2"
                        poster_path="test2"
                        release_date={2008}
                        starColor="#f2f2f2"
                        title="test2"
                        vote_average={1}
            />
</div>,
          <button
            className="more-btn"
            onClick={[Function]}
>
            View More
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(MovieCard)
                movie_id={0}
                overview="test1"
                poster_path="test1"
                release_date={2008}
                starColor="#f2f2f2"
                title="test1"
                vote_average={1}
/>,
              <Connect(MovieCard)
                movie_id={1}
                overview="test2"
                poster_path="test2"
                release_date={2008}
                starColor="#f2f2f2"
                title="test2"
                vote_average={1}
/>,
            ],
            "className": "movie-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "movie_id": 0,
                "overview": "test1",
                "poster_path": "test1",
                "release_date": 2008,
                "starColor": "#f2f2f2",
                "title": "test1",
                "vote_average": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "movie_id": 1,
                "overview": "test2",
                "poster_path": "test2",
                "release_date": 2008,
                "starColor": "#f2f2f2",
                "title": "test2",
                "vote_average": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View More",
            "className": "more-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View More",
          "type": "button",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`MovieList should match the snapshot with no movies 1`] = `"<main><div class=\\"movie-cards\\"></div><button class=\\"more-btn\\">View More</button></main>"`;

exports[`MovieList should match the snapshot with one movie 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieList
    favorites={
        Array [
            Object {
              "movie_id": 0,
              "overview": "test",
              "poster_path": "test",
              "release_date": 2008,
              "title": "test",
              "vote_average": 1,
            },
          ]
    }
    movies={
        Array [
            Object {
              "movie_id": 0,
              "overview": "test",
              "poster_path": "test",
              "release_date": 2008,
              "title": "test",
              "vote_average": 1,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="movie-cards"
>
          <Connect(MovieCard)
                    movie_id={0}
                    overview="test"
                    poster_path="test"
                    release_date={2008}
                    starColor="#f2f2f2"
                    title="test"
                    vote_average={1}
          />
</div>,
        <button
          className="more-btn"
          onClick={[Function]}
>
          View More
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(MovieCard)
              movie_id={0}
              overview="test"
              poster_path="test"
              release_date={2008}
              starColor="#f2f2f2"
              title="test"
              vote_average={1}
/>,
          ],
          "className": "movie-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "movie_id": 0,
              "overview": "test",
              "poster_path": "test",
              "release_date": 2008,
              "starColor": "#f2f2f2",
              "title": "test",
              "vote_average": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "View More",
          "className": "more-btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "View More",
        "type": "button",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="movie-cards"
>
            <Connect(MovieCard)
                        movie_id={0}
                        overview="test"
                        poster_path="test"
                        release_date={2008}
                        starColor="#f2f2f2"
                        title="test"
                        vote_average={1}
            />
</div>,
          <button
            className="more-btn"
            onClick={[Function]}
>
            View More
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(MovieCard)
                movie_id={0}
                overview="test"
                poster_path="test"
                release_date={2008}
                starColor="#f2f2f2"
                title="test"
                vote_average={1}
/>,
            ],
            "className": "movie-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "movie_id": 0,
                "overview": "test",
                "poster_path": "test",
                "release_date": 2008,
                "starColor": "#f2f2f2",
                "title": "test",
                "vote_average": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "View More",
            "className": "more-btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "View More",
          "type": "button",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
